web/packages/teleterm/src/mainProcess/agentDownloader/agentDownloader.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { pipeline } from 'node:stream/promises';
+import { createReadStream } from 'node:fs';
+import { join } from 'node:path';
+import { createUnzip } from 'node:zlib';
+import { exec } from 'node:child_process';
+import { promisify } from 'node:util';
+
+import { extract } from 'tar-fs';
+
+import { compareSemVers } from 'shared/utils/semVer';
+
+import Logger from 'teleterm/logger';
+
+import { RuntimeSettings } from '../types';
+
+import type { IFileDownloader } from './fileDownloader';
+
+const TELEPORT_CDN_ADDRESS = 'https://cdn.teleport.dev';
+const logger = new Logger('agentDownloader');
+
+interface AgentBinaryParams {
+  version: string;
+  platform: string;
+  arch: string;
+}
+
+/**
+ * Downloads and unpacks the agent binary, if it has not already been downloaded.
+ * If `RuntimeSettings` does not contain the correct version (so it equals 1.0.0-dev),
+ * the version from AGENT_VERSION env var is taken or the latest available version of the agent is downloaded.
```

Comment:
This comment does a good job of describing the function in a vacuum. I think we could add more details to it that are specific to our build process.

We could mention that `settings.appVersion` is set to a real version only for packaged apps that went through our CI build pipeline. In local builds, both for the development version and for packaged apps, `settings.appVersion` is set to `1.0.0-dev`.

In those cases, we fetch latest available stable version of the agent. `AGENT_VERSION` is available as an escape hatch for cases where we want to fetch a different version.

Reply:
Done